exchanges/kline/kline.go

```diff
@@ -459,20 +410,17 @@ func (h *IntervalRangeHolder) HasDataAtDate(t time.Time) bool {
 		return false
 	}
 	for i := range h.Ranges {
-		if tu >= h.Ranges[i].Start.Ticks && tu <= h.Ranges[i].End.Ticks {
-			for j := range h.Ranges[i].Intervals {
-				if tu >= h.Ranges[i].Intervals[j].Start.Ticks && tu < h.Ranges[i].Intervals[j].End.Ticks {
-					return h.Ranges[i].Intervals[j].HasData
-				}
-				if j == len(h.Ranges[i].Intervals)-1 {
-					if tu == h.Ranges[i].Start.Ticks {
-						return h.Ranges[i].Intervals[j].HasData
-					}
-				}
+		if tu < h.Ranges[i].Start.Ticks || tu >= h.Ranges[i].End.Ticks {
```

Comment:
I will have to review this, you make a good point. I believe it was something to do with it not being needed.

Reply:
So before it was `tu <= h.Ranges[i].End.Ticks` so if TU == End.Ticks it would actually mean the next candles open. So it might have turned up a positive value when It should not have. 

Also this was never hit because it was already being caught by the if statement above :

```go
if j == len(h.Ranges[i].Intervals)-1 {
	if tu == h.Ranges[i].Start.Ticks {
		return h.Ranges[i].Intervals[j].HasData
	}
}
```